stdlib/stomp/src/main/java/org/ballerinalang/stdlib/stomp/DefaultStompClient.java

```diff
@@ -0,0 +1,188 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.stomp;
+
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.bre.bvm.CallableUnitCallback;
+import org.ballerinalang.connector.api.*;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.services.ErrorHandlerUtils;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinalang.stdlib.stomp.StompConstants.*;
+
+/**
+ * Register stomp listener service.
+ *
+ * @since 0.990.2
+ */
+public class DefaultStompClient extends StompClient {
+    private CallableUnitCallback callableUnit;
+    private Map<String, Service> serviceRegistry = new HashMap<>();
+    private boolean connected;
+    private static final Logger log = LoggerFactory.getLogger(DefaultStompClient.class);
+
+    public DefaultStompClient(URI uri) {
+        super(uri);
+    }
+
+    @Override
+    public void onConnected(String sessionId) {
+        this.connected = true;
+
+        if (callableUnit != null) {
+            callableUnit.notifySuccess();
+        }
+    }
+
+    public boolean isConnected(){
+        return this.connected;
+    }
+
+    public void setCallableUnit(CallableUnitCallback callableUnit) {
+        this.callableUnit = callableUnit;
+    }
+
+    @Override
+    public void onDisconnected() {
+    }
+
+    @Override
+    public void onMessage(String messageId, String body, String destination) {
+            Resource messageResource = getMessageResource();
+            ProgramFile programFile = messageResource.getResourceInfo().getPackageInfo().getProgramFile();
+            BMap<String, BValue> msgObj = BLangConnectorSPIUtil.createBStruct(programFile, STOMP_PACKAGE, MESSAGE_OBJ);
+            List<ParamDetail> paramDetails = messageResource.getParamDetails();
+            String callerType = paramDetails.get(0).getVarType().toString();
+
+            if (callerType.equals("string")) {
+                    Executor.submit(messageResource, new ResponseCallback(body, messageId), new HashMap<>(), null, new BString(body));
+            } else if (callerType.equals("ballerina/stomp:Message")) {
+                    msgObj.addNativeData(STOMP_MSG, body);
+                    msgObj.put(MSG_CONTENT_NAME, new BString(body));
+                    msgObj.put(MSG_DESTINATION, new BString(destination));
+                    msgObj.put(MSG_ID, new BString(messageId));
+                    msgObj.addNativeData(CONFIG_FIELD_CLIENT_OBJ, this);
+                    Executor.submit(messageResource, new ResponseCallback(body, messageId), new HashMap<>(), null, msgObj);
+            }
```

Comment:
We need to discuss databinding. Shall we discuss in a future meeting?

Reply:
Sure will do